### PR TITLE
fixed parsing responses bodies, see spotify/ramlfications#12

### PR DESCRIPTION
**Pull request by jenner**
- The responses bodies map is now accepting more then just
  'application/json' media types. If there's no valid media
  type, the default one is used and the body spec is
  inspected for 'schema' and 'example' parameters
- Added XSDs for thingy and thingy-list
- Added parser tests wrt response bodies parsing

### CODE REVIEW

**Comment by econchick** [comment]
Hey @jenner! Thank you so much! I'll review this PR this week; nice catch!

File: ramlfications/parser.py [hunk 1]
```diff
@@ -927,24 +928,50 @@ def resp_headers(headers):
...
-                form_params=None,
-                config=root.config
-            )
-            body_list.append(b)
+            default_body = {}
+            for (key, spec) in body.items():
+                if key not in MEDIA_TYPES:
+                    # if a root mediaType was defined, the response body
+                    # may omit the mime_type definition
+                    if key == 'schema':
```
Comment (econchick) [inline ramlfications/parser.py]: I see you may have noticed the RAML spec's inconsistencies too :) 

It [says](http://raml.org/spec.html#body) that the `body` param should be a key/value map, where the key is a valid media type.  But in some cases - especially within the examples they use elsewhere in the spec, as well as some [examples](https://github.com/mulesoft/api-console/tree/master/dist/examples) they publish, media types are completely omitted.  

I think handling this the way you did is the pragmatic approach, and hopefully when they release 1.0 of the spec, they'll clear it up a bit.

**Comment by econchick** [comment]
Still reviewing this - but just a couple of things:

1 I would add/update a test in `test_parser.py` that `parser.py` returns what you expected. For example, when parsing the response schema/example in lines 515-521, you actually get (for the `example`) an array (I think? not too familiar with xml/xsd) of thingies, named `Foo` and `Bar`.  Essentially similar to the JSON example [here](https://github.com/spotify/ramlfications/blob/521b8a00ee2212df34d2999c6ecd2012adf39b3f/tests/test_parser.py#L106).

2. The `!!null` use case isn't tested - mind adding a test for that too?

**Comment by jenner** [comment]
yep, can do

**Comment by econchick** [comment]
Merged locally - thank you!

**Comment by econchick** [comment]
Also FYI will probably upload new release today or tomorrow - cheers!

**Comment by econchick** [comment]
FYI included in [release 0.1.5](https://github.com/spotify/ramlfications/releases/tag/0.1.5) and available on [PyPI](https://pypi.python.org/pypi/ramlfications)

**Pull request merged**